_pages/dev/ssr/server-side-rendering-contextual-logging.md

```diff
@@ -0,0 +1,244 @@
+```
+The following is the representation of the log message in the monitoring tool:
+![Log Representation In Kibana](assets/images/../../../../../assets/images/ssr/contextual_logging_kibana_logs.png)
+
+
+
+## Enabling the SSR Contextual Logging
+
+To enable the SSR contextual logging, add the `logger` flag to the configuration object as the second parameter to the method that decorates `ngExpressEngine` in the `server.ts` file. The following is an example:
+
+```ts
+import { ngExpressEngine as engine } from '@nguniversal/express-engine';
+import { NgExpressEngineDecorator } from '@spartacus/setup/ssr';
+
+[...]
+
+const ngExpressEngine = NgExpressEngineDecorator.get(engine, {
+  //add logger flag here
+  logger: true
+});
+
+[...]
+
+```
+
+From now on, your application will use the `DefaultExpressServerLogger` and provide logging context for all server-side rendered requests
+
+## Customizing the SSR Contextual Logging
+
+The Composable Storefront offers developers the ability to customize the SSR Contextual Logging. This includes the option to provide a custom logger in place of the default one. 
+
+To implement a custom logger, the developer must create a class that implements the `ExpressServerLogger` interface. Alternatively, they may extend the `DefaultExpressServerLogger` class. 
+
+Once the custom logger class is created, it can be passed to the `logger` property in the configuration object. 
+
+An example of how to do this is shown below:
+
+```ts
+import { ngExpressEngine as engine } from '@nguniversal/express-engine';
+import { NgExpressEngineDecorator } from '@spartacus/setup/ssr';
+import { CustomExpressServerLogger } from './custom-express-server-logger';
+
+[...]
+
+const ngExpressEngine = NgExpressEngineDecorator.get(engine, {
+  // pass custom logger here
+  logger: new CustomExpressServerLogger()
+});
+
+[...]
+```
+
+The following is an example of a custom logger implementation:
+
+```ts
+import { ExpressServerLogger } from '@spartacus/setup/ssr';
+
+export class CustomExpressServerLogger implements ExpressServerLogger {
+
+  //custom implementation of the `log` method
+  log(message: string, context: ExpressServerLoggerContext): void {
+    console.log({ message: `[My Custom LOG]: ${message}`, ...context });
+  }
+
+  [...]
+}
+```
+
+An important aspect of our solution is its extensibility. We have designed our solution to be compatible with enterprise logging libraries used for logging in platform-server such as [Winston](https://www.npmjs.com/package/winston) or [Pino](https://www.npmjs.com/package/pino), making it easy to add their features to our system.
+
+The following is an example of how to implement the custom logger with the `Pino` library:
```

Comment:
Let's move the disclaimer from line 157 to here:

`
**Disclaimer: This is only an example of how to use a third-party logger library. Do not treat this as a recommendation of the particular library.**
`